var/spack/repos/builtin/packages/py-melissa-core/package.py

```diff
@@ -31,17 +33,22 @@ class PyMelissaCore(PythonPackage):
     depends_on("py-setuptools@46.4:", type=("build"))
     # requirements.txt (SA)
     depends_on("py-pyzmq@22.3.0:", type="run")
-    depends_on("py-mpi4py@3.1.3:", type="run")
+    depends_on("py-mpi4py@3.1.3", type="run")
     depends_on("py-numpy@1.21:", type="run")
     depends_on("py-jsonschema@4.5:", type="run")
     depends_on("py-python-rapidjson@1.8:", type="run")
```

Comment:
```suggestion
    depends_on("py-python-rapidjson@1.8:", when="@develop,joss", type="run")
    depends_on("py-python-rapidjson@1.9:", when="@sc23", type="run")
```

Reply:
This is the result of a similar trick. The idea was to avoid updating the `python-rapidjson` library which only goes up to 1.8 for Spack. I have created [this PR](https://github.com/spack/spack/pull/39182) to add the two latest versions and make this commit work.